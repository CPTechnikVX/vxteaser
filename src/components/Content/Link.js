import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';

/**
 * General link component
 */
export default class Link extends React.PureComponent {
	render() {
		const {children, color, link, modifier} = this.props;
		const classList                         = [];
		let onClickFn;

		if (color) {
			classList.push(Constants.PrefixClassName.Color + color);
		}

		if (modifier) {
			classList.push(modifier);
		}

		if (link) {
			onClickFn = (e) => {
				e.preventDefault();
				e.stopPropagation();

				this.props.onLinkFn(link);
			};
		}

		return <a className={classnames(classList)} onClick={onClickFn}>{children}</a>;
	}
}

Link.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	color:    PropTypes.string,
	link:     PropTypes.string,
	modifier: PropTypes.string,
	/** @ignore */
	onLinkFn: PropTypes.func,
};
