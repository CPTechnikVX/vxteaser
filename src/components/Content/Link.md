```js
<Link link="https://www.example.com">
  Text of the link
</Link>
```
