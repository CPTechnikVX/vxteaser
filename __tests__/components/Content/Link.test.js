import React     from 'react';
import {shallow} from 'enzyme';
import Link      from '../../../src/components/Content/Link';

describe('render link', () => {
	test('with content', () => {
		const wrapper = shallow(
			<Link color="highlight" link="http://www.example.com" modifier="js-link">Link content</Link>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
