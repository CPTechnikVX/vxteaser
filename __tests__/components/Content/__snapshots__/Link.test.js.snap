// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render link with content 1`] = `
<a
  className="vxteaser-color--highlight js-link"
  onClick={[Function]}
>
  Link content
</a>
`;
